src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -155,64 +159,61 @@ applyCmd logger pactDbEnv miner gasModel pd spv cmdIn mcache
     buyGasEnv = CommandEnv Nothing Transactional pactDbEnv logger
       freeGasEnv pd' spv nid
 
-    apply = do
+    fatal = internalError
+
+    applyBuyGas = do
 
       buyGasResultE <- catchesPactError $! buyGas buyGasEnv cmd miner supply mcache
 
       case buyGasResultE of
 
-        Left e1 ->
-          jsonErrorResult buyGasEnv requestKey e1 [] (Gas 0) mcache
-          "tx failure for requestKey when buying gas"
-        Right (Left e) ->
-          -- this call needs to fail hard if Left. It means the continuation did not process
-          -- correctly, and we should fail the transaction
-          internalError e
-        Right (Right (T3 pactId buyGasLogs mcache')) -> do
-          logDebugRequestKey logger requestKey "successful gas buy for request key"
+        Left e1 -> fatal $ "tx failure for requestKey when buying gas: " <> tShow e1
```

Comment:
debugging this without logging has been a frustrating pain. Do you you mind logging this error as well?

Reply:
I'll add logging to `fatal`